src/main/java/seedu/address/model/student/Student.java

```diff
@@ -24,7 +25,7 @@ public class Student {
 
     // Data fields
     private final Set<Tag> tags = new HashSet<>();
-    private final Optional<String> group;
+    private final Optional<Group> group;
```

Comment:
Recommend reverting back to String for now.

Might have issues with loading data from json I think. Or logic of data loading needs to be updated.

Can choose to not revert if you tested that it works.

Reply:
When there are students added into groups, need to check if data is successfully loaded when reopening the AddressBook

Reply:
The data loading logic has been updated to match the changes in the student class. 
The check for data reloading will be done after the add student to group function is implemented.